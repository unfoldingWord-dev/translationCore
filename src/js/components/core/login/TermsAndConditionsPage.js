import React, { Component } from 'react';

class TermsAndConditionsPage extends Component {
  render() {
    return (
      <div style={{color: "var(--reverse-color)", display: "flex", flexDirection: "column", justifyContent: "center", alignItems: "center"}}>
        <h4 style={{marginTop: "40px"}}>Terms and Conditions</h4>
        <div>
          <p style={{padding: "15px", opacity: "0.8"}}>
            By creating a Door43 account you affirm that you have read
            and understand the terms and conditions and agree to:
          </p>
          <ul style={{marginLeft: "50px", padding: "10px", opacity: "0.8"}}>
            <li>
              Only add content to Door43 that does not infringe on someone else's copyright.
            </li><br />
            <li>
              Only add content to Door43 that does not conflict with the&nbsp;
              <span style={{cursor: "pointer", textDecoration: "underline"}}>
                Stament of Faith.
              </span>
            </li><br />
            <li>
              Release your contributions to the content on Door43's website under a&nbsp;
              <span style={{cursor: "pointer", textDecoration: "underline"}}>
                Creative Commons Attribution-ShareAlike 4.0 International License.
              </span>
            </li>
          </ul>
        </div><br /><br />
        <button
          className="btn-second"
          onClick={() => this.props.showLocalUserScreen(false)}>
          Go Back
        </button>
      </div>
    );
  }
}

export default TermsAndConditionsPage;
