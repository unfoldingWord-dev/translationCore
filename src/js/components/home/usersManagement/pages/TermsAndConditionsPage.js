import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon } from 'react-bootstrap';

const TermsAndConditionsPage = ({onBackClick, onFaithClick, onCreativeClick, translate}) => (
  <div>
    <button
      className="btn-second"
      onClick={onBackClick}>
      <Glyphicon glyph="share-alt" style={{transform: 'scaleX(-1)'}}/>&nbsp;
      {translate('go_back')}
    </button>
    <div style={{
      color: 'black',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center'
    }}>
      <h4 style={{marginTop: '40px'}}><b>Terms and Conditions</b></h4>
      <div>
        <p style={{padding: '15px'}}>
          By continuing, you affirm that you have read and understand the
          terms and conditions and agree to:
        </p>
        <ul style={{marginLeft: '50px', padding: '10px'}}>
          <li>
            Only add content that does not infringe on someone else&apos;s
            copyright.
          </li>
          <br/>
          <li>
            Only add content that does not conflict with the&nbsp;
            <span
              onClick={onFaithClick}
              style={{cursor: 'pointer', textDecoration: 'underline'}}>
              Statement of Faith.
            </span>
          </li>
          <br/>
          <li>
            Release your contributions to the content under a&nbsp;
            <span
              onClick={onCreativeClick}
              style={{cursor: 'pointer', textDecoration: 'underline'}}>
              Creative Commons Attribution-ShareAlike 4.0 International License.
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
);

TermsAndConditionsPage.propTypes = {
  translate: PropTypes.func.isRequired,
  onBackClick: PropTypes.func.isRequired,
  onFaithClick: PropTypes.func.isRequired,
  onCreativeClick: PropTypes.func.isRequired
};

export default TermsAndConditionsPage;
