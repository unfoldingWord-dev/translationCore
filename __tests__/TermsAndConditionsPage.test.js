/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import TermsAndConditionsPage from '../src/js/components/home/usersManagement/pages/TermsAndConditionsPage';

beforeAll(() => {
  configure({adapter: new Adapter()});
});

test('snapshot', () => {
  const wrapper = renderer.create(
    <TermsAndConditionsPage translate={key => key}
                            onFaithClick={jest.fn()}
                            onCreativeClick={jest.fn()}
                            onBackClick={jest.fn()}/>
  );
  expect(wrapper).toMatchSnapshot();
});

describe('callbacks', () => {
  configure({adapter: new Adapter()});
  const backCallback = jest.fn();
  const faithCallback = jest.fn();
  const creativeCallback = jest.fn();
  let wrapper = shallow(
    <TermsAndConditionsPage translate={key => key}
                            onFaithClick={faithCallback}
                            onCreativeClick={creativeCallback}
                            onBackClick={backCallback}/>
  );
  beforeEach(() => {
    backCallback.mockReset();
    faithCallback.mockReset();
    creativeCallback.mockReset();
  });

  test('back', () => {
    wrapper.find('button').simulate('click');
    expect(backCallback).toBeCalled();
    expect(faithCallback).not.toBeCalled();
    expect(creativeCallback).not.toBeCalled();
  });

  test('creative', () => {
    wrapper.find('#creative-link').simulate('click');
    expect(creativeCallback).toBeCalled();
    expect(backCallback).not.toBeCalled();
    expect(faithCallback).not.toBeCalled();
  });

  test('faith', () => {
    wrapper.find('#faith-link').simulate('click');
    expect(faithCallback).toBeCalled();
    expect(creativeCallback).not.toBeCalled();
    expect(backCallback).not.toBeCalled();
  });
});
