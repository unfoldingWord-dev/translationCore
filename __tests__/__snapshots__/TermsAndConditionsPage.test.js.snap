// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <button
    className="btn-second"
    onClick={[Function]}
  >
    <span
      className="glyphicon glyphicon-share-alt"
      style={
        Object {
          "transform": "scaleX(-1)",
        }
      }
    />
     
    go_back
  </button>
  <div
    style={
      Object {
        "alignItems": "center",
        "color": "black",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <h4
      style={
        Object {
          "marginTop": "40px",
        }
      }
    >
      <b>
        Terms and Conditions
      </b>
    </h4>
    <div>
      <p
        style={
          Object {
            "padding": "15px",
          }
        }
      >
        By continuing, you affirm that you have read and understand the terms and conditions and agree to:
      </p>
      <ul
        style={
          Object {
            "marginLeft": "50px",
            "padding": "10px",
          }
        }
      >
        <li>
          Only add content that does not infringe on someone else's copyright.
        </li>
        <br />
        <li>
          Only add content that does not conflict with the 
          <span
            id="faith-link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "textDecoration": "underline",
              }
            }
          >
            Statement of Faith.
          </span>
        </li>
        <br />
        <li>
          Release your contributions to the content under a 
          <span
            id="creative-link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "textDecoration": "underline",
              }
            }
          >
            Creative Commons Attribution-ShareAlike 4.0 International License.
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
